src/main/webapp/app/exercises/shared/exercise/exercise.utils.ts

```diff
@@ -196,14 +196,16 @@ export const participationStatus = (exercise: Exercise, testRun?: boolean): Part
     return ParticipationStatus.INACTIVE;
 };
 
-/**
- * The start exercise button should be available for programming exercises when
- * - there is no due date
- * - now is before the due date
- * - test run after due date is deactivated and manual grading is deactivated
- */
 export const isStartExerciseAvailable = (exercise: ProgrammingExercise): boolean => {
-    return exercise.dueDate == undefined || dayjs().isBefore(exercise.dueDate!);
+    return exercise.dueDate === undefined || dayjs().isBefore(exercise.dueDate);
```

Comment:
```suggestion
    return !exercise.dueDate || dayjs().isBefore(exercise.dueDate);
```

Reply:
I dislike falsyness and try to avoid it at all costs

Reply:
These two snippets have slightly different behavior:
![grafik](https://user-images.githubusercontent.com/26540346/197340358-2f75b41a-bc99-4944-9c46-0b672e5c50ee.png)

That's why I thing using falsy is better in this case.